domain/customers/repository.go

```diff
@@ -0,0 +1,93 @@
+package customers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"sync"
+
+	"github.com/nuts-foundation/nuts-demo-ehr/domain"
+)
+
+type Repository interface {
+	FindByID(id string) (*domain.Customer, error)
+	All() ([]domain.Customer, error)
+}
+
+type flatFileRepo struct {
+	filepath string
+	mutex    sync.Mutex
+	// records is a cache
+	records map[string]domain.Customer
+}
+
+func NewFlatFileRepository(filepath string) *flatFileRepo {
+	f, err := os.OpenFile(filepath, os.O_RDONLY, 0666)
+	defer f.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	return &flatFileRepo{
```

Comment:
maybe call `ReadAll()` here as well (fail fast) Otherwise you only notice your data is corrupt after interacting with the UI.